src/source/geojson_source.js

```diff
@@ -170,7 +170,7 @@ class GeoJSONSource extends Evented implements Source {
         const options = util.extend({}, this.workerOptions);
         const data = this._data;
         if (typeof data === 'string') {
-            options.url = resolveURL(data);
+            options.request = this.map._transformRequest(resolveURL(data));
```

Comment:
Is this the only place where the type isn't provided? Native [uses the `Source` type here](https://github.com/mapbox/mapbox-gl-native/blob/175e7e26e2008d1e55664bd1064536c23b6de86a/src/mbgl/style/sources/geojson_source.cpp#L53); let's do the same and then make this parameter required.